.gitmodules

```diff
@@ -0,0 +1,9 @@
+[submodule "third_party/gloo"]
+	path = third_party/gloo
+	url = https://github.com/facebookincubator/gloo
```

Comment:
nit-picking/question
What will happen when
- fuser is not `PROJECT_IS_TOP_LEVEL`
- I run `USE_DISTRIBUTED=0 python setup.py install`

?

Reply:
That's a good question!

When fuser is not top level build, so we are basically building from Pytorch where nvfuser `CMakeLists.txt` is added.
In that case, we do NOT need any submodule inside nvfuser and we'll just use gloo provided by pytorch.

When we switch the upstream pytorch `third_party/nvfuser` to a submodule, I don't know if there's any way for us to stop a totally unnecessary recursive submodule update.

Reply:
I'm a bit concerned with the possibility that we could have the divergent Gloo submodules whose side effects are unclear to me (partly because of my lack of knowledge)

ah, you've added the TODO for this: https://github.com/NVIDIA/Fuser/pull/59/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR192-R193

Reply:
> Gloo submodules whose side effects are unclear to me (partly because of my lack of knowledge)

I don't think I know better than you. And there is a possibility that any library we link against might differ from what upstream is using. Frankly speaking I don't have a good solution, other than sync'ing our submodule commit with upstream pytorch. :cry: